### PR TITLE
Trying to fix unittests

### PR DESCRIPTION
**Pull request by scott-lewis-nsidc**

### CODE REVIEW

**Comment by juliacollins** [comment]
Sadly, I don't have enough Javascript build knowledge to know why the build is failing. Keeping my fingers crossed that the javascript Slack channel can help.

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 language: node_js
+node_js:
+  "12.4.0"
```
Comment (juliacollins) [inline .travis.yml]: I see that the LTS version ("recommended for most users") is 10.16.0. Should we back off from the bleeding edge?

Comment (scott-lewis-nsidc) [reply .travis.yml]: Hmm, this is a good question.  I don't remember why I went with 12.4.0.  I can try bumping down to 10.16.0 and just make sure it still works ok.

File: src/scripts/views/result_item/SpatialMetadataView.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
   function (UtilityFunctions,
             spatialMetadataTemplate,
             spatialMetadatWithoutDataTemplate,
-            SpatialThumbnailView) {
+            SpatialThumbnailView,
+            sprintfjs
+            // _s
+  ) {
+
+  var sprintf = sprintfjs.sprintf;
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply src/scripts/views/result_item/SpatialMetadataView.js]: Basically, the underscore sprintf method is being deprecated, and the underscore.string package recommends using the sprintf-js package instead.
https://github.com/epeli/underscore.string/blob/master/sprintf.js
(If you use the underscore.string package, I think it was working, but spit out that warning when running tests, and presumably in the browser console as well)

File: src/templates/index-layout-BACKUP.jade [hunk 3]
```diff
@@ -0,0 +1,52 @@
+doctype
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply src/templates/index-layout-BACKUP.jade]: Oops, no.  I will remove this.

File: src/scripts/views/result_item/SpatialMetadataView.js [hunk 4]
```diff
@@ -1,11 +1,19 @@
 define(['lib/utility_functions',
-  'vendor/requirejs/text!templates/result_item/spatial_metadata.html',
-  'vendor/requirejs/text!templates/result_item/spatial_metadata_without_data.html',
-  'views/result_item/SpatialThumbnailView'],
+  'text!templates/result_item/spatial_metadata.html',
+  'text!templates/result_item/spatial_metadata_without_data.html',
+  'views/result_item/SpatialThumbnailView',
+  'sprintf'
+  // 'underscore.string'
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply src/scripts/views/result_item/SpatialMetadataView.js]: Will remove.

File: Gruntfile.js [hunk 5]
```diff
@@ -344,6 +271,71 @@ module.exports = function (grunt) {
       }
     },
 
+    // NOTE: The following contrib libraries had to have an id inserted into their
+    // js files so that requirejs wouldn't throw a fit about "anonymous" defines
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply Gruntfile.js]: I agree.  I tried to figure out what was causing it, but ultimately the only way I could get it to work was to get a local copy and manually edit it to give the module a "name".  The requireJS documents weren't very helpful in trying to track down WHY:
https://requirejs.org/docs/errors.html

The only thing I can think of is that somehow the test framework may be trying to include the files twice or something, or mixing requirejs stuff with other things.  It is definitely worth trying to fix, but I think the new story you created will suffice for now.

File: Gruntfile.js [hunk 6]
```diff
@@ -47,9 +22,9 @@ module.exports = function (grunt) {
     bundleExec: true,
     compass: true,
     precision: 2,
-    sourcemap: true,
+    // sourcemap: true,
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply Gruntfile.js]: I will remove them.  I think the sourcemap and lineNumbers values are no longer used in the updated sass grunt configuration (if I recall, it was throwing errors or warnings about unknown settings?), but I don't think we need them anymore.

File: .travis.yml (hunk 1 above)
Comment (scott-lewis-nsidc) [reply .travis.yml]: Seems to still work OK, so I'll use 10.16.0 instead.

File: package.json [hunk 7]
```diff
@@ -3,23 +3,42 @@
...
-    "grunt-contrib-requirejs": "^0.4.3",
-    "grunt-contrib-sass": "~0.7.3",
-    "grunt-contrib-watch": "~0.6.1",
-    "grunt-githooks": "^0.3.1",
-    "grunt-release": "^0.10.0",
-    "grunt-git": "~0.3.0",
-    "grunt-scss-lint": "0.1.11",
-    "grunt-shell": "^1.1.1",
-    "grunt-template-jasmine-requirejs": "git://github.com/iantruslove/grunt-template-jasmine-requirejs.git"
+    "connect-ssi": "1.1.1",
```
[the remark this answers is not in the document]
Comment (scott-lewis-nsidc) [reply package.json]: Can't remember why I changed this.  I reverted back to the NSIDC one and it seems to be working OK.  So this one is moot now.

**Pull request merged**